inference-engine/src/transformations/src/transformations/common_optimizations/nearest_neighbor_umpsampling_fusion.cpp

```diff
@@ -0,0 +1,322 @@
+        if (!reshape_1_node || reshape_1_node->get_input_partial_shape(0).is_dynamic()) return false;
+
+        uint64_t input_rank = static_cast<uint64_t>(reshape_1_node->get_input_partial_shape(0).rank().get_length());
+        const auto mul_const_shape = mul_const_node->get_output_shape(0);
+        const auto scales = get_scales_from_mul_const_shape(mul_const_shape, input_rank);
+        if (scales.empty() || std::all_of(scales.begin(), scales.end(), [](float s) { return s == 1.0f;})) { return false; }
+
+        const auto mul_const_value = mul_const_node->cast_vector<float>();
+        if (std::any_of(mul_const_value.begin(), mul_const_value.end(), [](float x){ return x != 1.0f; })) { return false; }
+
+        const auto concat_1_node = std::dynamic_pointer_cast<opset8::Concat>(reshape_1_node->input_value(1).get_node_shared_ptr());
+        if (!concat_1_node) return false;
+
+        const auto input_shape = reshape_1_node->get_input_shape(0);
+        if (!check_concat_1(concat_1_node, input_shape)) return false;
+
+        const auto concat_2_node = std::dynamic_pointer_cast<opset8::Concat>(pattern_to_output.at(concat_2).get_node_shared_ptr());
+        if (!concat_2_node) return false;
+
+        const auto new_spatial_shape = get_new_spatial_shape_from_concat_2(concat_2_node, input_shape);
+        if (new_spatial_shape.empty()) return false;
+
+        const auto ss_before_concat_1 = std::dynamic_pointer_cast<opset8::StridedSlice>(concat_1_node->input_value(0).get_node_shared_ptr());
+        const auto ss_before_concat_2 = std::dynamic_pointer_cast<opset8::StridedSlice>(concat_2_node->input_value(0).get_node_shared_ptr());
+        if (!ss_before_concat_1 || !ss_before_concat_2 || ss_before_concat_1.get() != ss_before_concat_2.get()) return false;
+
+        const auto shapeof_node = std::dynamic_pointer_cast<opset8::ShapeOf>(ss_before_concat_1->input_value(0).get_node_shared_ptr());
+        if (!shapeof_node) return false;
+
+        const auto node_before_shapeof = shapeof_node->input_value(0).get_node_shared_ptr();
+        const auto node_before_reshape_1 = reshape_1_node->input_value(0).get_node_shared_ptr();
+        if (node_before_shapeof.get() != node_before_reshape_1.get()) return false;
+
+        opset8::Interpolate::InterpolateAttrs attrs;
+        attrs.mode = opset8::Interpolate::InterpolateMode::NEAREST;
+        attrs.shape_calculation_mode = opset8::Interpolate::ShapeCalcMode::SCALES;
+        attrs.nearest_mode = opset8::Interpolate::NearestMode::ROUND_PREFER_FLOOR;
+        attrs.pads_begin = std::vector<size_t>{0};
+        attrs.pads_end = std::vector<size_t>{0};
+        attrs.antialias = false;
+        attrs.coordinate_transformation_mode = opset8::Interpolate::CoordinateTransformMode::HALF_PIXEL;
+        attrs.cube_coeff = -0.75f;
+
+        const auto scales_node = opset8::Constant::create(element::f32, {scales.size()}, scales);
```

Comment:
Should we take precision from input? or it is always FP32?

Reply:
Fixed.